using new iterator based API, this matches exactly the changes made upstream as of vmware-modules-304.2
--- a/vmnet-only/userif.c	2015-02-07 03:54:17.000000000 +0300
+++ c/vmnet-only/userif.c	2015-02-24 03:58:06.043605137 +0300
@@ -523,7 +523,15 @@
       .iov_base = buf,
       .iov_len  = len,
    };
-   return skb_copy_datagram_iovec(skb, 0, &iov, len);
+
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
+	return skb_copy_datagram_iovec(skb, 0, &iov, len);
+#else
+	struct iov_iter to;
+	iov_iter_init(&to, READ, &iov, 1, len);
+	return skb_copy_datagram_iter(skb, 0, &to, len);
+#endif
 }
 
 
